### PR TITLE
compiler(rime): add support

### PR DESCRIPTION
**Pull request by Freed-Wu**

### CODE REVIEW

File: runtime/compiler/rime_deployer.vim [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+" Android, NixOS, GNU/Linux, BSD
+for s:shared_data_dir in ['/sdcard/rime-data', '/run/current-system/sw/share/rime-data', '/usr/local/share/rime-data', s:prefix . '/share/rime-data']
+  if isdirectory(s:shared_data_dir)
+    break
+  endif
+endfor
+execute 'CompilerSet makeprg=rime_deployer\ --build\ %:p:h\' s:shared_data_dir
+
+let &cpoptions = s:save_cpoptions
+unlet s:save_cpoptions
```
Comment (chrisbra) [inline runtime/compiler/rime_deployer.vim:30]: unlet s:prefix s:shared_data_dir

Comment (Freed-Wu) [reply runtime/compiler/rime_deployer.vim:30]: Fixed

File: runtime/compiler/rime_deployer.vim [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+if s:prefix == v:null
+  let s:prefix = '/usr'
+endif
+" Android, NixOS, GNU/Linux, BSD
+for s:shared_data_dir in ['/sdcard/rime-data', '/run/current-system/sw/share/rime-data', '/usr/local/share/rime-data', s:prefix . '/share/rime-data']
+  if isdirectory(s:shared_data_dir)
+    break
+  endif
+endfor
+execute 'CompilerSet makeprg=rime_deployer\ --build\ %:p:h\' s:shared_data_dir
```
Comment (Konfekt) [inline runtime/compiler/rime_deployer.vim]: Likely `%:p:h:S` is more robust

Comment (Freed-Wu) [reply runtime/compiler/rime_deployer.vim]: Done.

**Comment by Freed-Wu** [comment]
Rebased.

**Pull request merged**